active/InputValidation/InputValidation.md

```diff
@@ -0,0 +1,572 @@
+<!-- Explanation and guidance that doesn't fit into the Examples section. -->
+
+<!-- APIs should only throw exceptions in exceptional conditions; basically,
+only when there's a bug in the caller, such as argument exception.  But if for some
+reason it's necessary for a caller to catch an exception from an API, call that
+out with an explanation either here or in the Examples -->
+
+# API Notes
+<!-- Option 1: Give a one or two line description of each API (type
+and member), or at least the ones that aren't obvious
+from their name.  These descriptions are what show up
+in IntelliSense. For properties, specify the default value of the property if it
+isn't the type's default (for example an int-typed property that doesn't default to zero.) -->
+
+<!-- Option 2: Put these descriptions in the below API Details section,
+with a "///" comment above the member or type. -->
+
+# API Details
+<!-- The exact API, in MIDL3 format (https://docs.microsoft.com/en-us/uwp/midl-3/) -->
+
+## InputPropertyAttribute
+The `InputPropertyAttribute` helps us ensure that we only generate validation code for the property specified by this attribute. This will help reduce code bloat by not generating code for properties that will never perform validation (like width). Controls can specify multiple properties as their input property. 
+
+    /// <summary>
+    /// Name of the property that is associated with user input.
+    /// </summary>
+    attribute InputPropertyAttribute
+    {
+         string Name;
+    }
+
+## InputValidationKind  
+Determines how the control displays validation visuals.
+
+    /// <summary>
+    ///    Determines how the control displays it's validation visuals
+    /// </summary>
+    [webhosthidden]
+    enum InputValidationKind
+    {
+        /// <summary>
+        /// Let's the Control dictate how to display visuals. This is the default value.
+        /// </summary>
+        Auto,
+        /// <summary>
+        /// An error icon displays to the right of the control that displays the error messages in a tool tip.
+        /// </summary>
+        Compact,
+        /// <summary>
+        /// Text for the error messages is displayed underneath the control.  
+        /// </summary>
+        Inline,
+        /// <summary>
+        /// Validation visuals are not displayed at all.
+        /// </summary>
+        Hidden,
+    };
+
+## IInputValidationControl
+Any control that wants to participate in input validation should derive from this interface. This, among other things, is what allows the markup compiler to know when to generate the code necessary for listening to the `INotifyDataErrorInfo.ErrorsChanged` event and propagate the results to the `IInputValidationControl.Errors` property.
+
+When the `ValidationCommand` property is non-null, the following will be true:
+1. When the `InputValidationKind` property is set to `Hidden`, the control will *not* invoke the `ValidationCommand.CanValidate` or `ValidationCommand.Validate` methods
```

Comment:
How would this work where a developer doesn't want errors indicated inline but in a list elsewhere in the view?

Reply:
Thanks for calling this out! We originally had a seperate enum called `InputValidationMode` that could be set to `Disabled`. Then `Hidden` would just mean "don't display" and we could display errors elsewhere like you suggest. I was trying to simplify the API as much as possible while retaining functionality, and this looks like something that was missed.